### ISSUE TITLE
Does not emit deprecated-method for unittest.case.TestCase.assertEquals

### ISSUE THREAD

**Issue by federicobond** [PyCQA/pylint#2226]
### Steps to reproduce

The following code should emit a `deprecated-method` warning for the use of `assertEquals`, which is an alias to `assertEqual`, according to the code [here](https://github.com/PyCQA/pylint/blob/56e8ec2ddcfd62a6b15baab1efa70ed3311a9a21/pylint/checkers/stdlib.py#L153).

````
import unittest

class Tests(unittest.TestCase):

    def test_foo(self):
        self.assertEquals(2 + 2, 4)
````

### Current behavior

Pylint does not emit any warnings. 

### Expected behavior

Pylint shoud emit a `deprecated-method` warning.

<hr>

I have tracked down the problem to the Call node inferring `unittest.case.TestCase.assertEqual` (note, without the "s") as its qname, which is not the one listed as deprecated. This is likely because assertEquals is just a copy of the assertEqual method assigned to another name. It should take into account the name under which it was actually called.

Unfortunately, I couldn't get a test case to fail with Pylint's test infrastructure. I tried adding the above code to `pylint/test/functional/deprecated_methods_py2.py` but tests seem to pass anyway.

**Comment by PCManticore** [comment]
Yes, this is somewhat sad, but unfortunately there's not a lot we can do without reengineering how the inference works. Currently when we infer a value, we go up to the leaf values, which in this case is just `assertEqual`. We can probably try to adapt `deprecated-method` to look for `assertEquals` explicitly (e.g. infer the object, then get its defining class and check to see if there is an alias for that deprecated name that points to the inferred value), which might be a lot easier than changing the inference.

**Comment by allanlewis** [comment]
Pylint now warns for a method named `assert_` on my own class: was this intended?
